src/visitor.ts

```diff
@@ -0,0 +1,116 @@
+import {parse, parser} from './parser';
+import {
+  AtomicExpressionNode,
+  ExpressionNode,
+  FromClauseNode,
+  RelationalOperatorNode,
+  SelectClauseNode,
+  SelectStatementNode,
+  WhereClauseNode,
+} from './nodes';
+import {
+  Expression,
+  FromClause,
+  ModelType,
+  RelationalOperator,
+  SelectClause,
+  SelectStatement,
+  WhereClause,
+} from './models';
+
+class SqlVisitor extends parser.getBaseCstVisitorConstructor() {
```

Comment:
This should probably be named `SQLAstBuilderVisitor` or something similar

The `visitor` part is an implementation detail, the important thing what it is actually used for.